libs/wingsdk/src/target-tf-aws/api.ts

```diff
@@ -0,0 +1,274 @@
+    ) {
+      fn = this.node.tryFindChild(
+        `${this.node.id}-OnGetRequest-${inflightNodeHash}`
+      );
+    } else {
+      const functionHandler = convertBetweenHandlers(
+        this,
+        `${this.node.id}-OnGetRequestHandler-${inflightNodeHash}`,
+        inflight,
+        join(__dirname, "api.ongetrequest.inflight.js"),
+        "ApiOnGetRequestHandlerClient"
+      );
+      fn = new cloud.Function(
+        this,
+        `${this.node.id}-OnGetRequest-${inflightNodeHash}`,
+        functionHandler,
+        props
+      );
+    }
+    if (!(fn instanceof Function)) {
+      throw new Error("Api only supports creating tfaws.Function right now");
+    }
+
+    this.api.addEndpoint(route, ApiMethods.GET, fn);
+
+    core.Resource.addConnection({
+      from: this,
+      to: fn,
+      relationship: "on_get_request",
+    });
+  }
+
+  /** @internal */
+  public _bind(): void {
+    throw new Error("Method not implemented.");
+  }
+
+  /** @internal */
+  public _toInflight(): Code {
+    throw new Error("Method not implemented.");
+  }
+}
+
+/**
+ * Represents a resource with an inflight "handle" method that can be passed to
+ * `Api.get`.
+ *
+ * @inflight `wingsdk.cloud.IApiGetRequestHandler`
+ */
+export interface IApiGetRequestHandler extends IResource {}
+
+export enum ApiMethods {
+  GET = "GET",
+  POST = "POST",
+  PUT = "PUT",
+  DELETE = "DELETE",
+  PATCH = "PATCH",
+  HEAD = "HEAD",
+  OPTIONS = "OPTIONS",
+}
+
+export class WingRestApi extends Construct {
+  public readonly api: ApiGatewayRestApi;
+  private redeployTrigger: Set<string>;
+  private deployment: ApiGatewayDeployment;
+  constructor(scope: Construct, id: string) {
+    super(scope, id);
+    this.redeployTrigger = new Set<string>();
+    this.api = new ApiGatewayRestApi(this, "api", {
+      name: ResourceNames.generateName(this, NAME_OPTS),
+    });
+
+    this.deployment = new ApiGatewayDeployment(this, "deployment", {
+      restApiId: this.api.id,
+      lifecycle: {
+        createBeforeDestroy: true,
+      },
+      triggers: {
+        redeployment: Lazy.stringValue({
+          produce: () => {
+            const value = createHash("sha256")
+              .update([...this.redeployTrigger].join(""))
+              .digest("hex");
+            console.log("redeployTrigger", value);
+            return value;
+          },
+        }),
+      },
+    });
+
+    // TODO: the deployment should depend on the methods, resources and integrations.
+    // this.deployment.dependsOn = Lazy.listValue({
+    //   produce: () => {
+    //     const deps: string[] = [];
+    //     console.log(this.node.children);
+    //     console.log("produce dependson");
+    //     for (const dep of this.redeployTrigger) {
+    //       deps.push(this.node.findChild(dep).node.id);
+    //     }
+    //     console.log("dependsOn", deps);
+    //     return deps;
+    //   },
+    // });
```

Comment:
Hey @RaphaelManke - yeah this seems like a tricky issue. From what I can see, it seems to me like CDKTF doesn't really like when you provide lazy list values (I tried a couple variations of your code and hit similar issues). I've pushed a commit that works around this by using an escape hatch - I think this is OK as long we can specify all dependencies ourselves.

Reply:
Ok your suggestion works (I had to add the resource type prior to the `friendlyUniqueId`). 
Now there is one small timing issue left, not sure if that can be solved. 

When a route is removed, terraform detects the change and deletes the integration, lambda etc. But this also needs a deployment. Since the dependent resources are deleted they are not part of the dependsOn. This results in the following behaviour. 
The deployment is created prior to deleting the resources from the apigateway. So the old resources are still part of the apigateway at the time the deployment is done. Since a deployment is a snapshot of the resources configured at this time the old resource is still part of the snapshot. 

Now one idea I have is to use the [sleep](https://registry.terraform.io/providers/hashicorp/time/latest/docs/resources/sleep) to wait a few seconds before creating a deployment. But this doesn't feel to be a good solution because it slows down the deployment by a fixed time value and does not guarantee the deployment always uses the new configuration because if the deletion takes longer than the configured sleep time it would still create a wrong api state afterwards.

Reply:
> The deployment is created prior to deleting the resources from the apigateway

Hmm -- I'm wondering if this is caused by the `create_before_delete` lifecycle option that is enabled on the deployment? I see that the Terraform docs recommends adding that field, but I wonder if it could be causing problems in the route-removal scenario...

I'm not sure of the best solution is at this moment. One idea I had from the terraform angle is that there is an option to set the "count" of a resource to 0, so that it's still in your template and can be depended on. But I don't know how we would generate this in Wing, since it would require remembering what resources you previously had in your app...

I'm okay with trying the sleep approach for now. I've posted a question on StackOverflow related to your issue so if I get an answer with a better solution I'll update you 🙏

Reply:
switching to the apispec approach fixes the problem 
https://github.com/winglang/wing/pull/1285#discussion_r1106407345